test/common/http/codec_impl_fuzz_test.cc

```diff
@@ -292,7 +295,8 @@ class HttpStream : public LinkedObject<HttpStream> {
       break;
     }
     case test::common::http::DirectionalAction::kTrailers: {
-      if (state.isLocalOpen() && state.stream_state_ == StreamState::PendingDataOrTrailers) {
+      if ((http_protocol_ > Protocol::Http11 || http1_enable_trailers_) && state.isLocalOpen() &&
```

Comment:
nit: Please add a TODO comment here (need to add http support1).

BTW: from the [style guide](https://github.com/envoyproxy/envoy/blob/main/STYLE.md#deviations-from-google-c-style-guidelines):
```
Use your GitHub name in TODO comments, e.g. TODO(foobar): blah.
```

Reply:
I think the change to the logic is allow trailers if either H2 or we have H1 settings enabling trailers anded with the existing logic

Reply:
I have added a comment to explain that trailers are allowed for http >= 2 or for http1x when the configuration is set. http1 support is present, but can only be used, when trailers are allowed in the configuration.
I hope the comment helps in understanding.

Reply:
I have added that corpus in commit 7f2bf68 .
I had to use the enable_trailers in the http1-client-config (see 2e37169), too, for it to get working. Is that ok?

Reply:
Thanks.
I think it would be better to add a new type `Http1ClientSettings` and add a corresponding field to [`Http1ClientServerSettings `](https://github.com/envoyproxy/envoy/blob/7f2bf68f0179e1ffa40633545bdcafeade692fa7/test/common/http/codec_impl_fuzz.proto#L97) (similar to [`Http2ClientServerSettings `](https://github.com/envoyproxy/envoy/blob/7f2bf68f0179e1ffa40633545bdcafeade692fa7/test/common/http/codec_impl_fuzz.proto#L110)).
This way a client can support trailers but the server doesn't, and vice-versa.

Reply:
I think this is an improvement from the current state, and can have independently  server / client  H1 trailers later. Might be worth opening an issue to track it.

Reply:
I have split up the client and server configs. Is that what you had in mind?